Mathlib/Algebra/MonoidAlgebra/Equiv.lean

```diff
@@ -0,0 +1,143 @@
+/-
+Copyright (c) 2022 Yuyang Zhao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yuyang Zhao
+-/
+
+import Mathlib.Algebra.Algebra.Tower
+import Mathlib.Algebra.MonoidAlgebra.Basic
+
+/-!
+TODO
+-/
+
+noncomputable section
+
+namespace AddMonoidAlgebra
+
+@[simps]
+def ringEquivCongrLeft {R S G : Type _} [Semiring R] [Semiring S] [AddMonoid G] (f : R ≃+* S) :
+    AddMonoidAlgebra R G ≃+* AddMonoidAlgebra S G :=
```

Comment:
This file should be easy to PR separately.

In general I think these `Equiv`s should be obtained by combining both directions (`RingHom`s in this case). We should show that `AddMonoidAlgebra · G` is a functor (prove the `comp` and `id` lemmas) and deduce that both directions are inverse to each other.

Some assumptions might be able to be generalized, e.g. here [AddZeroClass G] should work.